bridges/AppleAppStoreBridge.php

```diff
@@ -78,94 +94,157 @@ public function getName()
+        try {
+            if (isset($data['attributes'])) {
+                $this->name = $data['attributes']['name'] ?? null;
+                $author = $data['attributes']['artistName'] ?? null;
+                $this->debugLog("Found app details in attributes: {$this->name} by {$author}");
+                return [$this->name, $author];
+            }
+            
+            $this->name = "App " . $this->getInput('id');
+            $this->debugLog("App details not found, using default: {$this->name}");
+            return [$this->name, "Unknown Developer"];
+        } catch (\Exception $e) {
+            $this->debugLog("Error extracting app details: " . $e->getMessage());
+            $this->name = "App " . $this->getInput('id');
+            return [$this->name, "Unknown Developer"];
         }
     }
 
-    public function collectData()
+    private function getVersionHistory($data)
     {
-        $id = $this->getInput('id');
-        $country = $this->getInput('country');
         $platform = $this->getInput('p');
-
-        switch ($platform) {
-            case 'mac':
-                $data = $this->getDataFromShoebox($id, $platform, $country);
-                break;
-
-            default:
-                $token = $this->getJWTToken($id, $platform, $country);
-                $data = $this->getAppData($id, $platform, $country, $token);
+        $this->debugLog("Extracting version history for platform: {$platform}");
+        
+        try {
+            $platform_key = self::PLATFORM_MAPPING[$platform] ?? $platform;
+            
+            if (isset($data['attributes']['platformAttributes'][$platform_key]['versionHistory'])) {
+                return $data['attributes']['platformAttributes'][$platform_key]['versionHistory'];
+            }
+            
+            $this->debugLog("No version history found for {$platform}");
+            return [];
+            
+        } catch (\Exception $e) {
+            $this->debugLog("Error extracting version history: " . $e->getMessage());
+            return [];
         }
+    }
 
-        $versionHistory = $this->getVersionHistory($data, $platform);
-        $name = $this->name = $data['attributes']['name'];
-        $author = $data['attributes']['artistName'];
-
-        foreach ($versionHistory as $row) {
-            $item = [];
-
-            $item['content'] = nl2br($row['releaseNotes']);
-            $item['title'] = $name . ' - ' . $row['versionDisplay'];
-            $item['timestamp'] = $row['releaseDate'];
-            $item['author'] = $author;
-
-            $item['uri'] = $this->makeHtmlUrl($id, $country);
-
-            $this->items[] = $item;
+    public function collectData()
+    {
+        try {
+            $this->debugLog("Getting data for " . $this->getInput('p') . " app");
+            $data = $this->getAppData();
+            
+            list($name, $author) = $this->extractAppDetails($data);
```

Comment:
pls dont use `list()`. I think php has proper lang syntax for this unpacking.

Something like:

```
[$name, $auth] = $this->extractAppDetails($data);
```